### PR TITLE
Enhance Docstrings in `aepsych/kernels`

### PR DESCRIPTION
**Pull request by yalsaffar**
One of several PRs addressing issue #366 to improve docstring coverage.

Improves documentation in `aepsych/kernels` for better clarity and consistency.
- Adds missing docstrings to functions and methods across kernel modules.
- Updates existing docstrings with refined type hints and a unified structure.

### CODE REVIEW

**Comment by JasonKChow** [comment]
Need rebase, let me know when this is rebased and I'll import it to internal.

**Review by JasonKChow — CHANGES_REQUESTED** [review]
Need rebase

**Review by JasonKChow — APPROVED** [review]

File: aepsych/kernels/pairwisekernel.py [hunk 1]
```diff
@@ -16,8 +16,13 @@ class PairwiseKernel(Kernel):
     """
 
     def __init__(
-        self, latent_kernel: Any, is_partial_obs: bool = False, **kwargs
+          self, latent_kernel: Any, is_partial_obs: bool=False, **kwargs
```
Comment (crasanders) [inline aepsych/kernels/pairwisekernel.py:19]: I guess I missed this before, but the type of latent_kernel should be Kernel

File: aepsych/kernels/rbf_partial_grad.py [hunk 2]
```diff
@@ -30,6 +30,21 @@ class RBFKernelPartialObsGrad(RBFKernelGrad):
     def forward(
         self, x1: torch.Tensor, x2: torch.Tensor, diag: bool = False, **params: Any
     ) -> torch.Tensor:
+        """Computes the covariance matrix between x1 and x2 based on the RBF
+        
+        Args:
+            x1 (torch.Tensor): A `b x n x d` or `n x d` tensor, where `d = 2k` and `k` is the dimension of the latent space.
+            x2 (torch.Tensor): A `b x m x d` or `m x d` tensor, where `d = 2k` and `k` is the dimension of the latent space.
+            diag (bool): Should the Kernel compute the whole covariance matrix or just the diagonal? Defaults to False.
```
Comment (crasanders) [inline aepsych/kernels/rbf_partial_grad.py:38]:
```suggestion
            diag (bool): Should the Kernel compute the whole covariance matrix (False) or just the diagonal (True)? Defaults to False.
```

**Review by crasanders — CHANGES_REQUESTED** [review]

**Pull request closed without merging**